sympy/physics/control/control_plots.py

```diff
@@ -959,3 +964,332 @@ def bode_plot(system, initial_exp=-5, final_exp=5,
+
+    See Also
+    ========
+
+    impulse_response_plot, ramp_response_plot ,step_response_plot
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Root_locus_analysis
+
+    """
+    pz = pole_zero_plot(system, show=False, grid=grid)
+    x, y = root_locus_numerical_data(system, k_max=k_max, num=num)
+    pz.plot(x, y)
+    pz.title('Root Loci')
+
+    if grid:
+        pz.grid(True)
+    if show_axes:
+        pz.axhline(0, color='black')
+        pz.axvline(0, color='black')
+    if show:
+        pz.show()
+        return
+
+    return pz
+
+
+def nichols_numerical_data(system, initial_omega=0.01, final_omega=100, **kwargs):
+    """
+    Returns the numerical data of Nichols plot of the system.
+    It is internally used by ``nichols_plot`` to get the data
+    for plotting Nichols plot. Users can use this data to further
+    analyse the dynamics of the system or plot using a different
+    backend/plotting-module.
+
+    Parameters
+    ==========
+
+    system : SISOLinearTimeInvariant
+        The system for which the pole-zero data is to be computed.
+    initial_omega : Number, optional
+        The initial value of frequency. Defaults to 0.01.
+    final_omega : Number, optional
+        The final value of frequency. Defaults to 100.
+
+    Returns
+    =======
+
+    tuple : (phase_points, mag_points)
+        phase_points = phase values of the Nichols plot.
+        mag_points = magnitude values of the Nichols plot.
+
+    Raises
+    ======
+
+    NotImplementedError
+        When a SISO LTI system is not passed.
+
+        When time delay terms are present in the system.
+
+    ValueError
+        When more than one free symbol is present in the system.
+        The only variable in the transfer function should be
+        the variable of the Laplace transform.
+
+    Examples
+    ========
+
+    >>> from sympy.abc import s
+    >>> from sympy.physics.control.lti import TransferFunction
+    >>> from sympy.physics.control.control_plots import nichols_numerical_data
+    >>> tf1 = TransferFunction(-(0.1)*s**3 - (2.4)*s**2 - 181*s - 1950, s**3 + (3.3)*s**2 + 990*s + 2600, s)
+    >>> nichols_numerical_data(tf1)   # doctest: +SKIP
+    (array([179.83501857, 179.67004337, 179.50508061, ..., 166.86071969,
+        166.86233751, 166.8639549 ]),
+    array([ -2.49883392,  -2.49901149,  -2.49930742, ..., -20.5300856,
+        -20.52996573, -20.52984591]))
+
+    See Also
+    ========
+
+    nichols_plot
+    """
+    _check_system(system)
+    expr = system.to_expr()
+
+    _w = Dummy("w", real=True)
+    repl = I*_w
+    w_expr = expr.subs({system.var: repl})
+
+    mag = 20*log(Abs(w_expr), 10)
+    phase = arg(w_expr)*180/pi
+
+    x = np.linspace(initial_omega, final_omega, 10000)
+
+    mag_func = lambdify(_w, mag, modules=['numpy'])
+    phase_func = lambdify(_w, phase, modules=['numpy'])
+    mag_points = mag_func(x)
+    phase_points = phase_func(x)
+
```

Comment:
This can be handled by the sympy's plotting module as follows - 
```
from sympy.plotting.plot import Parametric2DLineSeries
phase_points, mag_points = Parametric2DLineSeries(phase,mag,(_w,initial_omega,final_omega), nb_of_points = 10000, adaptive= False).get_points()
return phase_points, mag_points
```
We will get rid of `np.linspace `and numerical `lambdify` with this.
The generation is slower but that is expected and accepted. The accuracy of plots w.r.t MALTAB is not affected - 
```
tf = TransferFunction(-(0.1)*s**3 - (2.4)*s**2 - 181*s - 1950, s**3 + (3.3)*s**2 + 990*s + 2600, s)
nichols_plot(tf,1,100)
```
![image](https://github.com/sympy/sympy/assets/90216905/efbe8570-140f-4c99-ad1f-4d7ee95a9ea3)

With this **nichols_plot** will not use numpy functionalitites. I'll send in a commit soon if no one opposes this.

Reply:
I will be reverting this commit. Because I realized recently that `Parametric2DLineSeries` does not work well with Python 3.9 / 3.8 versions as you can see. As SymPy works with any version above 3.8 for now, I would not expect a user to know that the accurate results will be obtained only on version 3.10.
To make sure that eventually the current method will should be replaced by `Parametric2DLineSeries`, I will raise an issue once this is ready to go in.
```
s = Symbol('s')
tf = TransferFunction(-(0.1)*s**3 - (2.4)*s**2 - 181*s - 1950, s**3 + (3.3)*s**2 + 990*s + 2600, s)
nichols_plot(tf,1,100)
tf1 = TransferFunction(s**2 + (0.1)*s + 7.5,s**4 + (0.12)*s**3 + 9*s**2,s)
nichols_plot(tf1)
```
**!python --version    3.10.12**

![image](https://github.com/sympy/sympy/assets/90216905/6de5cb37-499d-445a-ad42-fadb15ba478d)
![image](https://github.com/sympy/sympy/assets/90216905/53fba273-af51-4110-a990-99ff2e81b1e4)

**!python --version    3.9.6**

<img width="478" alt="image" src="https://github.com/sympy/sympy/assets/90216905/c1a2cf11-9a1b-4226-961e-eb8937247190">
<img width="479" alt="image" src="https://github.com/sympy/sympy/assets/90216905/7a72f7ff-6074-4794-b6e5-24adbf01c122">